meson.build

```diff
@@ -256,6 +257,7 @@ executable(
   sources,
   dependencies: [ libnvme_dep, libuuid_dep, json_c_dep, libz_dep,
                   libhugetlbfs_dep ],
+  link_args: '-ldl',
```

Comment:
btw, link_args is only used when building shared binaries. So nothing to worry about static builds.